### ISSUE TITLE
My Findings on 0.6.0 on Raspberry Pi 3B+

### ISSUE THREAD

**Issue by GaWr26** [InstaPy/InstaPy#4823]
In general it looks really good and seemed to work in the beginning. Unfortunately, after a couple of runs, I got blocked on two of three accounts I'm using this on.

Also my session freezes for quite some time on my Raspberry Pi. It freezes so bad, that the Raspberry becomes unresponsive. No more ssh or VNC access. I have to reset the power to boot back up again. I figured out that if I just wait, sometimes even a couple of hours, the script continues and the raspberry becomes available again. 
The console output looks normal and doesn't report any issues.
Any idea why this is happening?

Also running multiple instances doesn't work. I start the first bot and it runs. Trying to start a second one, firefox crashes before it can load a page.
The first instance isn't affected by the crash of the second one. 
With chrome I could run three accounts simultaneously without a problem.
```
Workspace in use: "/home/pi/InstaPy"
Traceback (most recent call last):
  File "bot.py", line 26, in <module>
    session = InstaPy(username=insta_username, password=insta_password, multi_logs=True, disable_image_load=False, headless_browser=False, split_db=True, geckodriver_path="assets/geckodriver")
  File "/home/pi/InstaPy/instapy/instapy.py", line 318, in __init__
    self.logger,
  File "/home/pi/InstaPy/instapy/browser.py", line 114, in set_selenium_local_session
    options=firefox_options,
  File "/home/pi/.local/lib/python3.7/site-packages/selenium/webdriver/firefox/webdriver.py", line 174, in __init__
    keep_alive=True)
  File "/home/pi/.local/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 157, in __init__
    self.start_session(capabilities, browser_profile)
  File "/home/pi/.local/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 252, in start_session
    response = self.execute(Command.NEW_SESSION, parameters)
  File "/home/pi/.local/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/home/pi/.local/lib/python3.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: connection refused
```

Here are some other general errors I am experiencing:

long pause in the script(freeze):
```
INFO [2019-08-08 09:06:46] [xxx]  User: 'xxx'  |> followers: 6675  |> following: 650  |> relationship ratio: 10.26
INFO [2019-08-08 09:19:58] [xxx]  --> Already following 'xxx'!
```

and some other errors here:
```
INFO [2019-08-09 09:35:18] [xxx]  User: 'hernan1035'  |> followers: 250  |> following: 889  |> relationship ratio: 0.28
INFO [2019-08-09 09:37:42] [xxx]  Timed out with failure while explicitly waiting until page fully loaded!

INFO [2019-08-09 10:37:59] [xxx]  Last follow is verified after reloading the page!
INFO [2019-08-09 10:37:59] [xxx]  --> Followed 'b'hernan1035''!
```

```
Traceback (most recent call last):
  File "bot.py", line 97, in <module>
    session.follow_user_followers(['user1','user2','user3','user4','user5','user6'], amount=25, randomize=True, sleep_delay=100)
  File "/home/pi/InstaPy/instapy/instapy.py", line 3477, in follow_user_followers
    validation, details = self.validate_user_call(person)
  File "/home/pi/InstaPy/instapy/instapy.py", line 1313, in validate_user_call
    self.logfolder,
  File "/home/pi/InstaPy/instapy/util.py", line 207, in validate_username
    browser, username, logger
  File "/home/pi/InstaPy/instapy/util.py", line 1057, in get_relationship_counts
    web_address_navigator(browser, user_link)
  File "/home/pi/InstaPy/instapy/util.py", line 1183, in web_address_navigator
    browser.get(link)
  File "/home/pi/.local/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 333, in get
    self.execute(Command.GET, {'url': url})
  File "/home/pi/.local/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/home/pi/.local/lib/python3.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 241, in check_response
    raise exception_class(message, screen, stacktrace, alert_text)
selenium.common.exceptions.UnexpectedAlertPresentException: Alert Text: None
Message: Dismissed user prompt dialog: A script on this page may be busy, or it may have stopped responding. You can stop the script now, or you can continue to see if the script will complete.

Script: chrome://global/content/bindings/tabbox.xml:405
```

Just pulled the latest changes and will run on the still working account. If I see any issues I will post them here.

Keep up the great work guys! Really appreciate it!

**Comment by GaWr26** [comment]
Update: Unfollowing 100 Users worked well. No fatal issues, no freeze. Just this:
```
INFO [2019-08-12 12:26:42] [xxx]  Ongoing Unfollow [50/92]: now unfollowing 'b'theuser''...
INFO [2019-08-12 12:32:15] [xxx]  Timed out with failure while explicitly waiting until page fully loaded!

ERROR [2019-08-12 12:37:25] [xxx]  Hey! Last unfollow is not verified out of an unexpected failure!
```

Next up: `follow_user_followers`

**Comment by maxxfly** [comment]
I try the 0.6.0 on Raspberry Pi 3B+ with without headless and with firefox, at start it is ok, I keep 100 Mo of memory, but some hours later, the raspberry is unresponsive because it swaps. 
I think that FF load too many memory. I think that by this way, the best solution is to move to a Raspberry 4 with 2 giga or 4 giga

**Comment by GaWr26** [comment]
@maxxfly I second that behavior. It does eventually come back though. Up to two hours of being unresponsive for me.
I opened a new issue here https://github.com/timgrossmann/InstaPy/issues/4839 as I discovered the freeze only happens on follow_user_followers. unfollow_users works like a charm.
I already ordered the Raspberry 4 with 4Gb but it's out of stock everywhere :(

**Comment by maxxfly** [comment]
I observed that on Firefox on Raspberry, the part where Instapy collects the likesr on my 4 latests photos can be very long. I think that it becomes complicated to analyze the webpage with a long "content" (I have only 50 linkers for theses pictures... it is not crazy)

**Comment by GaWr26** [comment]
I enabled swap now with 500MB and it seems to run much better.
Got all 100 follows in with no freeze.

What I did:
`sudo nano /etc/dphys-swapfile`
there I set:
`CONF_SWAPSIZE=500`

Then activate swapping: `sudo systemctl start dphys-swapfile`
In addition have it start automatically: `sudo systemctl enable dphys-swapfile`

You can then check for memory using `free`
```
              total        used        free      shared  buff/cache   available
Mem:         948304      638472      134740       25824      175092      229560
Swap:        511996      204032      307964
```

Update: Also running two accounts simultaneously works with swap enabled!

**Comment by maxxfly** [comment]
Have you not a risk to kill your micro sd with the swap? In my case the logs are saved on my nas by nfs to avoid to use too much the micro sd card

**Comment by GaWr26** [comment]
> Have you not a risk to kill your micro sd with the swap? In my case the logs are saved on my nas by nfs to avoid to use too much the micro sd card

Well, it's probably not the best treatment for the SD card but to be honest, I don't care. I have several cards and it only needs to last until the Raspberry 4 with 4GB arrives which should happen in the next two weeks.

For now, I'm back in business and made decent progress yesterday. Everything ran through and worked as expected.

**Comment by breuerfelix** [comment]
@GaWr26 what is your progress in the past 2 weeks where I was on vacation ? any updates on blocks etc ?

**Comment by GaWr26** [comment]
Hey @breuerfelix, now it was my time for vacation and I just got back :)
I had three accounts running and two of them got blocks on follows. One did actually continue quite well with 100 follows/day. Unfortunately just before my vacation this account also got blocked on follows. Unfollows and Likes work. I paused following on this account now. 
The other two accounts where I had the follow block I didn't run the bot for two or even three weeks and started again last Friday. One of them now runs with 15 follows/day and the other one got blocked again after two days with 5follows/hour and 15 follows/day.
Things are not looking good for me and I'm kinda stuck. Missing the days where we could do 600 follows/day without any blocks...
I hope things will get better once the elections in the US is over. I believe the new IG approach has to do with this and they want to avoid another cambridge analytica scandal.
Lately I didn't have time to read through issues and news on InstaPy but I believe everyone has these issues, right? Is there a post somewhere where all this is summarized? What are your plans for InstaPy?

As always, thanks for all the great work you put into this!
